### PR TITLE
chore(bhc): change get to async

### PR DESCRIPTION
**Pull request by xingoxu**
Use Async XMLHttpRequest to get a better experience with Ukagaka

### CODE REVIEW

File: binota/bangumi-homepage-calendar.user.js [hunk 1]
```diff
@@ -44,14 +44,24 @@ const DEFAULT_CONFIG = {
...
+  return new Promise((resolve, reject) => {
+    var sync = (typeof sync === 'undefined') ? true : sync;
+    var req = new XMLHttpRequest();
+    req.onload = () => {
+      if (req.status === 200)
+        return resolve(req.responseText);
+      else
+        return reject();  
+    };
+    req.open('GET', url, !sync);
```
Comment (BinotaLIU) [inline binota/bangumi-homepage-calendar.user.js]: It seems to be a little confused. We passed a sync argument, but the bahavior in the function seems always use syncronouse request.
Can you just rewrite whole `get` function, and use `fetch()` as a replacement for `Promise + XMLHttpRequest`?

File: binota/bangumi-homepage-calendar.user.js [hunk 2]
```diff
@@ -432,9 +444,10 @@ var Application = function() {
   var checkBreakPoint = (lastUpdate) ? (new Date(lastUpdate - NSECS_IN_DAY)) : (new Date(now.getTime() - (NSECS_IN_DAY * (DAYS_IN_YEAR + 1))));
 
   //Check
-  var getPage = function(get, calendar, tag) {
-    for(let i = 1; ; i++) {
-      var page = get(i);
+  var getPage = async function(get, calendar, tag) {
+    for (let i = 1; ; i++) {
+      // wait some time let Ukagaka show the message completely;
+      var page = await Promise.all([get(i), wait(200)]).then(([resultA, noResult]) => resultA);
```
Comment (BinotaLIU) [inline binota/bangumi-homepage-calendar.user.js]:
```javascript
var page = await wait(200).then(() => get(i));
```
Maybe this is better, I think.. 🤔

**Comment by xingoxu** [comment]
I am a little busy lately and I will verify and amend it tomorrow.
I'm really sorry for your waiting.

File: binota/bangumi-homepage-calendar.user.js (hunk 2 above)
Comment (xingoxu) [reply binota/bangumi-homepage-calendar.user.js]: Let 200ms and the request in parallel is no problem.

**Comment by xingoxu** [comment]
If we use fetch API, the sync request will not be supported any longer.

File: binota/bangumi-homepage-calendar.user.js (hunk 2 above)
Comment (BinotaLIU) [reply binota/bangumi-homepage-calendar.user.js]: I see. Maybe I misunderstood that.

**Comment by BinotaLIU** [comment]
😅

**Pull request merged**